server/sonar-server/src/main/java/org/sonar/server/es/EsClientProvider.java

```diff
@@ -56,13 +56,14 @@ public EsClient provide(Settings settings) {
         esSettings.put("client.transport.sniff", true);
         nativeClient = TransportClient.builder().settings(esSettings).build();
         Arrays.stream(settings.getStringArray(ProcessProperties.CLUSTER_SEARCH_HOSTS))
-          .map(Host::parse)
-          .forEach(h -> h.addTo(nativeClient));
+          .map(HostAndPort::fromString)
```

Comment:
ok, I understand. I missed that the target of the change was really `sonar.cluster.search.hosts` when cluster is enabled and the current node has ES.

But I think I still have  a point.

Only it applies to when clustering is not enabled and search is configured with properties `ProcessProperties.SEARCH_HOST` and `ProcessProperties.SEARCH_PORT`. Also, it's not a regression nor a bug, but rather an **inconsistency**: 
* when clustering is enabled, "no port" means "use the default port" while
* when clustering is not enabled, "no port" means "bind to any port" (car port 0 => attribution automatique)

This should be discussed with @christophelevis